flight_phase_files/HALO/EUREC4A_HALO_Flight-Segments_20200215.yaml

```
@@ -159,6 +159,7 @@ segments:
   irregularities:
   - ascend from 10.4 km to 11.2 km at circle start
   - SAM HALO-0215_s32 sonde added manually, released before traffic diversion
+  - One sonde HALO-0215_s44 overlaps with next segment (HALO-0215_s6) as per plan
```

I'd argue that **SAM** is also a correct tag for this irregularity. The "as per plan" is kind of redundant, as otherwise you wouldn't manually change the sonde association and I think we didn't add this to previous comments as well.
```suggestion
  - SAM sonde HALO-0215_s44 overlaps with next segment (HALO-0215_s6)
```

---

Hmm... But the `SAM` tag for HALO-0215_s6 needs to be there for only one circle, because it will by definition belong to the other, and I think in this case, it will belong naturally to the timestamps of this circle stamp. I can remove the "as per plan" part and rephrase it as "as decided before the flight". The "as per plan" was to indicate that it was intentionally planned for the flight to launch the sonde thinking of it as an overlap for both segments. Some sondes that we attribute as `SAM` were not part of the plan, but during the segmentation, we realise that we need to include the sondes in the previous or next segment.

---

I also need to change the circle segment ids from an `s` to a `c`.

---

I think we used "ci" for the P3 circles. Two characters gave us more flexibility than one.

---

@Geet-George your argument regarding the `SAM` seems to be reasonable. And I like the "as decided before the flight" as well.
@RobertPincus I think, Geet referred only to the segment name in the comment which seems to be a typo. I would otherwise strongly encourage not to change any segment ids once they are established as chances are quite high that those ids have persisted in inputs and outputs of other peoples scripts already. But I don't see any reason why "ci" shouldn't be equally fine. Doing textual comparisons on the ids isn't a good idea anyway, so no one should rely on the fact that it is either `c` or `ci`.

---

Ahh.. I am sorry. I might have overlooked that then. I don't remember the circle IDs in P3 circles being `ci` instead of `c`. If `ci` gives more flexibility, I am up for it. But then maybe we should change it also for all HALO flights? So, unless someone says otherwise, I'll change all `circle` segment ids to say `ci` instead of `c`.

---

As said just before, I would **strongly** encourage **not** to change any segment ids once they are established.
My reasoning is as follows: the main purpose of an identifier is to uniquely identify a certain thing. Not more and not less. If those identifiers change over time, we loose exactly that property. E.g. someone (person or algorithm) may still use the old identifier and another one uses the new one and both might not realize that they are talking about the same thing.
Things which explicitly should not be goals for those identifier are:
* optimization for human readability
* possibility to derive information about what is stored in the attributes of the identified object

I can see that there is some desire to partly support these non-goals as well (sha256 sums are really hard to remember), but they should never defeat the main purpose. Also having a guideline of how to create unique ids is helpful.

Accordingly, it is perfectly fine to use whatever fits best for generating those ids (i.e. `c` for HALO and `ci` for P3 or even a  mix of those), but changing them is not a good idea.

PS: My personal issue with changing those ids is also related to the above: I already spent a lot of (computing-) time to produce maps of the specMACS data, which are per-segment. So I already kind of rely on the idea that the ids stay constant in time.